1-js/02-first-steps/09-comparison/article.md

```diff
@@ -109,108 +109,108 @@ alert( Boolean(b) ); // true
+A razão é que uma verificação de igualdade `==` e comparações `> < >= <=` funcionam de forma diferente. As comparações convertem `null` em um número, tratando-o como `0`. É por isso que (3) `null >= 0` é verdadeiro e (1) `null > 0` é falso.
 
-On the other hand, the equality check `==` for `undefined` and `null` is defined such that, without any conversions, they equal each other and don't equal anything else. That's why (2) `null == 0` is false.
+Por outro lado, a verificação de igualdade `==` para `undefined` e `null` é definida de tal forma que, sem nenhuma conversão, eles são iguais e não são iguais a mais nada. É por isso que (2) `null == 0` é falso.
 
-### An incomparable undefined
+### Um incomparável indefinido
 
-The value `undefined` shouldn't be compared to other values:
+O valor `undefined` não deve ser comparado a outros valores:
 
 ```js run
 alert( undefined > 0 ); // false (1)
 alert( undefined < 0 ); // false (2)
 alert( undefined == 0 ); // false (3)
 ```
 
-Why does it dislike zero so much? Always false!
+Por que não gostam do zero? Sempre falso!
 
-We get these results because:
+Obtemos esses resultados porque:
 
-- Comparisons `(1)` and `(2)` return `false` because `undefined` gets converted to `NaN` and `NaN` is a special numeric value which returns `false` for all comparisons.
-- The equality check `(3)` returns `false` because `undefined` only equals `null`, `undefined`, and no other value.
+- Comparações `(1)` e `(2)` retornam `false` porque `undefined` é convertido em `NaN` e `NaN` é um valor numérico especial que retorna `false` para todas as comparações.
+- A verificação de igualdade `(3)` retorna `false` porque `undefined` só é igual a `null`, `undefined` e nenhum outro valor.
 
-### Avoid problems
+### Evite problemas
 
-Why did we go over these examples? Should we remember these peculiarities all the time? Well, not really. Actually, these tricky things will gradually become familiar over time, but there's a solid way to avoid problems with them:
+Por que examinamos esses exemplos? Devemos nos lembrar dessas peculiaridades o tempo todo? Bem, na verdade não. Na verdade, essas coisas complicadas gradualmente se tornarão familiares com o tempo, mas há uma maneira sólida de evitar problemas com elas:
 
-- Treat any comparison with `undefined/null` except the strict equality `===` with exceptional care.
-- Don't use comparisons `>= > < <=` with a variable which may be `null/undefined`, unless you're really sure of what you're doing. If a variable can have these values, check for them separately.
+- Trate qualquer comparação com `undefined/null` exceto a igualdade estrita `===` com cuidado excepcional.
+- Não use comparações `>= > < <=` com uma variável que pode ser `null/undefined`, a menos que você tenha certeza do que está fazendo. Se uma variável puder ter esses valores, verifique-os separadamente.
```

Comment:
```suggestion
- Não use comparações `>= > < <=` com uma variável que pode ser `null/undefined`, a menos que você tenha a certeza do que está fazendo. Se uma variável puder vir a ter esses valores, verifique-os separadamente.
```

Reply:
Eu uso uma extensão ("LTeX") que me auxilia na sintaxe e concordância do texto.

A expressão `vir a ter` foi identificada como uma perífrase, podendo ser simplificada.

![image](https://github.com/javascript-tutorial/pt.javascript.info/assets/35238611/fc380654-4b36-402c-a3aa-11c30bdbf141)

Não há necessidade do `vir a`, vide exemplificação do motivo:

https://community.languagetool.org/rule/show/VIR_A_VERBO_VERBO?lang=pt-BR

![image](https://github.com/javascript-tutorial/pt.javascript.info/assets/35238611/d21ee297-95de-499d-a928-c080689b5f7e)

Substitui o trecho por outra expressão que deixa mais nítido, tudo bem?

```diff
- ... Se uma variável puder a ter esses valores ...
+ ... Se uma variável tiver `null` ou `undefined` ...

Reply:
@jonnathan-ls eu recomendo usar a expressão *Se uma variável for `null` ou `undefined`*.